frontend/lib/src/WidgetStateManager.ts

```diff
@@ -283,6 +283,26 @@ export class WidgetStateManager {
     }
   }
 
+  /* Sometimes users change an input field and directly click on a button to trigger a rerun
+   *  the setTimeout is used to ensure the input field value is sent to the server before the button click
+   *  by putting the the button click in the end of the event loop
+   *
+   * Returns a promise that is resolved as soon as the timeout was triggered for testing purposes.
```

Comment:
I don't fully understand this part -- is it referring to the 0 timeout that will ensure we don't wait to add this to the task queue, or something else?

Reply:
Are you referring to the entire JS doc or the last part about the promise return? I have updates the doc string slightly, perhaps its clearer now?

Reply:
Yeah, the rewording is clearer to me, thanks! Added a nit on the english.